### PR TITLE
SONARIAC-1566 Docker parser should consider Exec form with characters behind as Shell form

### PR DESCRIPTION
**Pull request by rudy-regazzoni-sonarsource**

### CODE REVIEW

File: iac-extensions/docker/src/main/java/org/sonar/iac/docker/parser/grammar/DockerLexicalConstant.java [hunk 1]
```diff
@@ -30,6 +30,7 @@ public class DockerLexicalConstant {
   public static final String STRING_LITERAL_WITHOUT_QUOTES = "(?:(?!" + LINE_BREAK + ")[^\\s])++";
   public static final String STRING_LITERAL_OLD = "(?:(?:" + STRING_LITERAL_WITH_QUOTES + ")|(?:" + STRING_LITERAL_WITHOUT_QUOTES + "))+";
   public static final String EQUALS_OPERATOR = "=";
+  public static final String RBRACKET_END_EXEC_FORM = "\\](?=[" + LexicalConstant.WHITESPACE + "]*(?:[\r\n]|$))";
```
Comment (mstachniuk) [inline iac-extensions/docker/src/main/java/org/sonar/iac/docker/parser/grammar/DockerLexicalConstant.java:33]: Please try if this variant works:
```suggestion
  public static final String RBRACKET_END_EXEC_FORM = "\\](?=[" + LexicalConstant.WHITESPACE + "]*+(?:[\r\n]|$))";
```

File: iac-extensions/docker/src/test/java/org/sonar/iac/docker/tree/impl/ExecFormImplTest.java [hunk 2]
```diff
@@ -30,10 +33,6 @@
 import org.sonar.iac.docker.tree.api.ExecForm;
 import org.sonar.iac.docker.tree.api.SyntaxToken;
 
-import static org.assertj.core.api.Assertions.assertThat;
```
Comment (mstachniuk) [inline iac-extensions/docker/src/test/java/org/sonar/iac/docker/tree/impl/ExecFormImplTest.java:33]: static imports should be here not before other imports 
https://github.com/SonarSource/sonar-developer-toolset

**Review by mstachniuk — APPROVED** [review]
LGTM

File: iac-extensions/docker/src/test/java/org/sonar/iac/docker/tree/impl/ExecFormImplTest.java (hunk 2 above)
Comment (rudy-regazzoni-sonarsource) [reply iac-extensions/docker/src/test/java/org/sonar/iac/docker/tree/impl/ExecFormImplTest.java:33]: I agree, I don't get why my intellij keep reorganizing them before :confused:

image

I tried several things including installing the last version, I'm unable to get it to properly organize the imports :expressionless:

Comment (rudy-regazzoni-sonarsource) [reply iac-extensions/docker/src/test/java/org/sonar/iac/docker/tree/impl/ExecFormImplTest.java:33]: Finally found a way by starting on a fresh **sonar-iac** project.

**Pull request merged**